File: convert_fonts.py
import base64
import os

# 获取脚本所在目录的绝对路径
script_dir = os.path.dirname(os.path.abspath(__file__))

# 字体文件路径
fonts = [
    {
        'name': 'Ola Sans Compact',
        'path': os.path.join(script_dir, 'fonts', 'OlaSansCompactVF.ttf')
    },
    {
        'name': 'Ola Sans Mono',
        'path': os.path.join(script_dir, 'fonts', 'Ola Sans Mono.ttf')
    }
]

def convert_font_to_base64(font_path):
    try:
        with open(font_path, 'rb') as f:
            return base64.b64encode(f.read()).decode()
    except Exception as e:
        print(f"Error reading font file {font_path}:", e)
        return None

def generate_font_face_css(font_name, base64_data):
    return f"""@font-face {{
    font-family: '{font_name}';
    src: url(data:font/truetype;charset=utf-8;base64,{base64_data}) format('truetype');
    font-weight: 300 800;
    font-style: normal;
    font-display: swap;
}}
"""

def main():
    css_content = ''
    
    for font in fonts:
        print(f"Processing font: {font['name']}")
        print(f"Font path: {font['path']}")
        base64_data = convert_font_to_base64(font['path'])
        if base64_data:
            css_content += generate_font_face_css(font['name'], base64_data)
            print(f"Successfully converted {font['name']}")
        else:
            print(f"Failed to convert {font['name']}")
    
    output_path = os.path.join(script_dir, 'fontfaces.css')
    with open(output_path, 'w') as f:
        f.write(css_content)
    print(f'Font faces have been converted and saved to {output_path}')

if __name__ == '__main__':
    main()

File: convertFonts.js
const fs = require('fs');
const path = require('path');

// 字体文件路径
const fonts = [
    {
        name: 'OlaSansCompactVF',
        path: './fonts/OlaSansCompactVF.ttf'
    },
    {
        name: 'OlaSansMono',
        path: './fonts/Ola Sans Mono.ttf'
    }
];

// 转换字体文件到Base64
function convertFontToBase64(fontPath) {
    try {
        const font = fs.readFileSync(fontPath);
        return font.toString('base64');
    } catch (error) {
        console.error(`Error reading font file ${fontPath}:`, error);
        return null;
    }
}

// 生成CSS @font-face 声明
function generateFontFaceCSS(fontName, base64Data) {
    return `@font-face {
    font-family: '${fontName}';
    src: url(data:font/truetype;charset=utf-8;base64,${base64Data}) format('truetype');
    font-weight: 300 800;
    font-style: normal;
    font-display: swap;
}\n`;
}

// 主函数
function main() {
    let cssContent = '';
    
    fonts.forEach(font => {
        const base64Data = convertFontToBase64(font.path);
        if (base64Data) {
            cssContent += generateFontFaceCSS(font.name, base64Data);
        }
    });

    // 将结果写入到文件
    fs.writeFileSync('fontfaces.css', cssContent);
    console.log('Font faces have been converted and saved to fontfaces.css');
}

main();
